src/plugins/ExtendedProfileSampling.h

```diff
@@ -0,0 +1,111 @@
+
+private:
+    // Control: general
+    float _binwidth;
+    unsigned long _startSampling;
+    unsigned long _writeFrequency;
+    unsigned long _stopSampling;
+    bool _singleComp;
+
+    // Higher moments
+    bool _sampleHigherMoms;
+
+    // Control: chemical potential
+    bool _sampleChemPot;
+    bool _lattice;
+    float _factorNumTest;
+    unsigned long _samplefrequency;
+
+    // Auxiliary variables
+    uint16_t _numBinsGlobal;
+    unsigned long _lenVector;
+    std::array<double, 3> _globalBoxLength;
+    double _slabVolume;
+    CellProcessor* _cellProcessor;
+    ParticlePairsHandler* _particlePairsHandler;
+    Molecule _mTest;
+
+    // Accumulated quantities over _writeFrequency per bin
+    // NOTE: Only the root process knows correct values (with exceptions)
+    std::vector<unsigned long> _numMolecules_accum;             // Number of molecules in bin
+    std::vector<double> _density_accum;                         // Local density
+    std::vector<double> _temperature_accum;                     // Temperature (drift corrected)
+    std::vector<double> _ekin_accum;                            // Kinetic energy
+    std::vector<double> _epot_accum;                            // Potential energy
+    std::vector<double> _pressure_accum;                        // Pressure
+    std::vector<double> _chemPot_accum;                         // Chemical potential as sampled in ms2 (Widom insertion method)
+    std::vector<unsigned long> _countNTest_accum;               // Number of inserted test particles for chem. pot. sampling
+    std::array<std::vector<double>, 3> _temperatureVect_accum;  // Kinetic temperature in each direction (drift corrected)
+    std::array<std::vector<double>, 3> _velocityVect_accum;     // Drift velocity in each direction
+    std::array<std::vector<double>, 3> _pressureVect_accum;     // Pressure in each direction
+    std::array<std::vector<double>, 3> _energyfluxVect_accum;   // Energy flux (heat flux plus enthalpy flux) in each direction
+
+    // Higher moments
+    std::vector<double> _hmDelta_accum;                               // Higher moment: Delta
+    std::array<std::vector<double>, 9> _hmPressure_accum;             // Higher moment: Pressure; cxcx, cxcy, cxcz, cycx, cycy, cycz, czcx, czcy, czcz
+    std::array<std::vector<double>, 9> _hmR_accum;                    // Higher moment: R; cxcx, cxcy, cxcz, cycx, cycy, cycz, czcx, czcy, czcz
+    std::array<std::vector<double>, 27> _hmM_accum;                    // Higher moment: M; cicxcx, cicxcy, cicxcz, cicycx, cicycy, cicycz, ciczcx, ciczcy, ciczcz mit i = x,y,z
+
+    std::vector<unsigned long> _countSamples;                   // Number of samples; can vary from bin to bin as some bins could be empty
+
+    void resizeVectors(); // Change size of accumulation vectors
+    void resetVectors();  // Set accumulation vectors to zero
+
+public:
+    ExtendedProfileSampling();
+	~ExtendedProfileSampling();
```

Comment:
```suggestion
	~ExtendedProfileSampling() override = default;
```
Then you can delete the trivial implementation in the `.cpp`